common/test/acceptance/pages/studio/asset_index.py

```diff
@@ -88,3 +93,371 @@ def return_results_set(self):
+            self.q(css=file_input_css).results[0].clear()
+
+        self.wait_for_element_visibility(
+            '.alert', 'Upload status alert is visible.')
+        self.wait_for_ajax()
+        self.wait_for_files_upload(len(file_names))
+
+    @wait_for_js
+    def wait_for_files_upload(self, number):
+        """
+        Wait for file(s) to upload.
+
+        Arguments:
+            number (int): number of uploaded files.
+        """
+        return EmptyPromise(
+            lambda: self.number_of_asset_files == number,
+            "Files finished uploading"
+        ).fulfill()
+
+    @property
+    @wait_for_js
+    def is_previous_button_disabled(self):
+        return 'disabled' in self.q(css=self.PAGINATION_PAGE_ELEMENT).first.attrs('class')[0]
+
+    @property
+    @wait_for_js
+    def is_next_button_disabled(self):
+        return 'disabled' in self.q(css=self.PAGINATION_PAGE_ELEMENT).nth(
+            self.number_of_pagination_buttons - 1).attrs('class')[0]
+
+    @property
+    @wait_for_js
+    def is_previous_button_on_page(self):
+        """Note: the two conditions cover when the button is and is not disabled."""
+        return 'previous' in self.q(css=self.PAGINATION_PAGE_ELEMENT).first.text \
+            or 'previous' in self.q(css=self.PAGINATION_PAGE_ELEMENT + ' span').first.text
+
+    @property
+    @wait_for_js
+    def is_next_button_on_page(self):
+        """Note: the two conditions cover when the button is and is not disabled."""
+        return 'next' in self.q(css=self.PAGINATION_PAGE_ELEMENT).nth(
+            self.number_of_pagination_buttons - 1).text \
+            or 'next' in self.q(css=self.PAGINATION_PAGE_ELEMENT + ' span').nth(
+                self.number_of_pagination_buttons - 1).text
+
+    @wait_for_js
+    def click_pagination_page_button(self, index):
+        """
+        Click pagination previous button.
+        Return False if no pagination page button at specified index.
+        """
+        self.wait_for_ajax()
+        if index < self.number_of_pagination_page_buttons:
+            self.q(css=self.PAGINATION_PAGE_ELEMENT + '.page-item').nth(index).click()
+            self.wait_for_ajax()
+            return True
+        return False
+
+    @wait_for_js
+    def click_pagination_next_button(self):
+        """
+        Click pagination next button.
+        Return False if next button disabled.
+        """
+        self.wait_for_ajax()
+        if not self.is_next_button_disabled:
+            self.q(css=self.PAGINATION_PAGE_ELEMENT).nth(
+                self.number_of_pagination_buttons - 1).click()
+            self.wait_for_ajax()
+            return True
+        return False
+
+    @wait_for_js
+    def click_pagination_previous_button(self):
+        """
+        Click pagination previous button.
+        Return False if previous button disabled.
+        """
+        self.wait_for_ajax()
+        if not self.is_previous_button_disabled:
```

Comment:
Nit: same "enabled" comment as above

Reply:
I don't see an above comment, but I'm assuming you're suggesting I change the method from `is_previous_button_disabled` to `is_previous_button_enabled`, right? I'll do that.